network-store-iidm-impl/src/main/java/com/powsybl/network/store/iidm/impl/TerminalBusViewImpl.java

```diff
@@ -29,20 +29,75 @@
         this.attributes = attributes;
     }
 
+    private boolean isNodeBeakerTopologyKind() {
+        return getVoltageLevelResource().getAttributes().getTopologyKind() == TopologyKind.NODE_BREAKER;
+    }
+
+    private boolean isBusBeakerTopologyKind() {
+        return getVoltageLevelResource().getAttributes().getTopologyKind() == TopologyKind.BUS_BREAKER;
+    }
+
+    private <T> AbstractTopology<T> getTopologyInstance() {
+        return isNodeBeakerTopologyKind() ?
+                (AbstractTopology<T>) NodeBreakerTopology.INSTANCE : (AbstractTopology<T>) BusBreakerTopology.INSTANCE;
+    }
+
+    private Resource<VoltageLevelAttributes> getVoltageLevelResource() {
+        return index.getVoltageLevel(attributes.getVoltageLevelId()).orElseThrow(IllegalStateException::new).getResource();
+    }
+
+    private Bus calculateBus() {
+        return isNodeBeakerTopologyKind() ?
+                getTopologyInstance().calculateBus(index, getVoltageLevelResource(), attributes.getNode(), false, true) :
+                getTopologyInstance().calculateBus(index, getVoltageLevelResource(), attributes.getBus(), false, true);
+    }
+
     @Override
     public Bus getBus() {
-        VoltageLevelImpl voltageLevel = index.getVoltageLevel(attributes.getVoltageLevelId()).orElseThrow(IllegalStateException::new);
-        Resource<VoltageLevelAttributes> voltageLevelResource = voltageLevel.getResource();
-
-        if (voltageLevelResource.getAttributes().getTopologyKind() == TopologyKind.NODE_BREAKER) {
-            return NodeBreakerTopology.INSTANCE.calculateBus(index, voltageLevelResource, attributes.getNode());
-        } else {
-            return BusBreakerTopology.INSTANCE.calculateBus(index, voltageLevelResource, attributes.getBus());
-        }
+        return calculateBus();
     }
 
     @Override
     public Bus getConnectableBus() {
-        throw new UnsupportedOperationException("TODO");
+        VoltageLevelImpl voltageLevel = index.getVoltageLevel(attributes.getVoltageLevelId()).orElseThrow(IllegalStateException::new);
+        if (isBusBeakerTopologyKind()) { // Merged bus
+            return voltageLevel.getBusView().getMergedBus(attributes.getConnectableBus());
+        } else { // Calculated bus
+            return findConnectableBus();
+        }
+    }
+
+    private Bus findConnectableBus() {
+        VoltageLevelImpl voltageLevel = index.getVoltageLevel(attributes.getVoltageLevelId()).orElseThrow(IllegalStateException::new);
+
+        final Bus[] foundBus = {getBus()};
```

Comment:
Because when you pass a parameter that represents a primitive type to a function
and you modify this parameter inside the function this modification 
this modification is not seen outside because a local copy is done

Reply:
Sorry, when you pass a parameter in a inner class, it must be declared final.
But i want modify this parameter in the inner class
`Cannot assign a value to final variable 'foundBus'`

Reply:
Ok, understood !